File: androidDemo/src/main/java/com/novage/p2pml/demo/ui/screens/player/PlayerUiState.kt
package com.novage.p2pml.demo.ui.screens.player

import com.novage.p2pml.api.models.PeerDetails
import com.novage.p2pml.demo.ui.screens.player.models.AvailableTracks

data class PlayerUiState(
    val isVideoReady: Boolean = false,
    val isP2PActive: Boolean = false,
    val fatalError: String? = null,
    val userMessage: String? = null,

    val totalDownloaded: Long = 0,
    val p2pDownloaded: Long = 0,
    val httpDownloaded: Long = 0,
    val uploadTotal: Long = 0,
    val peers: List<PeerDetails> = emptyList(),

    val availableTracks: AvailableTracks = AvailableTracks()
)

File: iosDemo/iosDemo/UI/Screens/Player/PlayerUiState.swift
import Foundation

struct PlayerUiState {
    var isVideoReady: Bool = false
    var isP2PActive: Bool = false
    var fatalError: String?
    var userMessage: String?

    var p2pDownloaded: Int64 = 0
    var httpDownloaded: Int64 = 0
    var uploadTotal: Int64 = 0
    var peers: [String] = []

    var availableTracks = AvailableTracks()
}
